physical-based-animations.go

```diff
@@ -90,53 +91,108 @@ func run() {
 	prevDt := 0.002
 
 	last := time.Now()
+
+	gui := GUI{
+		win: win,
+	}
+
+	gui.CreateBatch("assets/sprites/spritesheet.png")
+
+	playButton := Button{
+		position:     pixel.V(10, 10),
+		croppingArea: pixel.R(0, 160, 80, 240),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandlePlayClick,
+	}
+
+	gui.NewButton(playButton)
+
+	pauseButton := Button{
+		position:     pixel.V(10, 100),
+		croppingArea: pixel.R(0, 80, 80, 160),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandlePauseClick,
+	}
+
+	gui.NewButton(pauseButton)
+
+	stopButton := Button{
+		position:     pixel.V(10, 190),
+		croppingArea: pixel.R(0, 0, 80, 80),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandleStopClick,
+	}
+
+	gui.NewButton(stopButton)
+
+	cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
+
+	win.SetMatrix(cam)
+
+	gui.SetMatrix(cam)
+	gui.BindState(&state)
+	gui.MainLoop()
+	gui.Draw()
+
 	for !win.Closed() {
-		dt := time.Since(last).Seconds()
-		last = time.Now()
-		timeElapsed += dt
 
-		cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
-		win.SetMatrix(cam)
-		win.Clear(colornames.Whitesmoke)
+		win.Update()
 
 		batch.Clear()
-		updateParticles(particleSystem.particles, batch, dt, cam)
-
-		batch.Draw(win)
-
-		for timeElapsed > timeForOneParticle {
-			pos := particleSystem.position
-			angle := (rand.Float64() - 0.5) * (particleSystem.angle * (math.Pi / 180))
-			speed := pixel.V(0, startSpeed).Rotated(angle)
-			nextPost := pos.Add(speed.Scaled(PixelsPerMeter).Scaled(prevDt)).Add(Gravity.Scaled(PixelsPerMeter).Scaled(prevDt * prevDt * 0.5))
-			// fmt.Printf("pos: %f %f	prev: %f %f\n", pos.X, pos.Y, prevPos.X, prevPos.Y)
-			particle := Particle{
-				position:     pos,
-				nextPosition: nextPost,
-				speed:        speed,
-				prevDt:       prevDt,
-				sprite:       *particleSprite,
-				lifespan:     100.0,
-				alive:        0.0,
-			}
-			particleSystem.particles = append(particleSystem.particles, particle)
-			timeElapsed = timeElapsed - timeForOneParticle
-		}
 
-		win.Update()
+		if !gui.GetState().paused && !gui.GetState().stopped {
```

Comment:
If the user decides to `pause`/`stop` this whole block should not be even considered. That is using something like this would be better imo

```go
for !win.Closed() {
    if !program.running {
        continue
    }
    ...
}
```

Reply:
For know I do not have any suitable solution for that. I will create an issue for that.